### PR TITLE
chore(notifications): refactor marketing notification

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- update deep link options
- update MarkingNotificationEvent to allow for different channels
- use updated marketingNotificationTrigger mutation in admin-panel

### CODE REVIEW

File: core/notifications/proto/notifications.proto [hunk 1]
```diff
@@ -237,19 +237,49 @@ message PriceChanged {
...
   optional DeepLink deep_link = 3;
+  bool should_send_push = 4;
+  bool should_add_to_history = 5;
+  bool should_add_to_bulletin = 6;
 }
 
-message LocalizedPushContent {
+message LocalizedContent {
   string title = 2;
   string body = 3;
```
Comment (thevaibhav-dixit) [inline core/notifications/proto/notifications.proto:249]: why 2, 3 instead of 1,2 .. . This is a breaking change so might as well fix this also

Comment (UncleSamtoshi) [reply core/notifications/proto/notifications.proto:249]: Yeah thats a good point, I will fix it

File: core/notifications/src/graphql/convert.rs [hunk 2]
```diff
@@ -10,10 +10,11 @@ impl From<history::StatefulNotification> for types::StatefulNotification {
         Self {
             deep_link: notification.deep_link().map(|d| d.to_link_string()),
             id: ID(notification.id.to_string()),
-            title: notification.message.title,
-            body: notification.message.body,
+            title: notification.message.title.clone(),
+            body: notification.message.body.clone(),
             created_at: types::Timestamp::from(created_at),
             acknowledged_at: acknowledeg_at.map(types::Timestamp::from),
+            add_to_bulletin: notification.add_to_bulletin(),
```
Comment (bodymindarts) [inline core/notifications/src/graphql/convert.rs:17]: move these above line 13 - then you can de-structure without needing to clone. Best to avoid clone where possible.
Before you start moving attributes you can call fns.

File: core/notifications/src/user_notification_settings/repo.rs [hunk 3]
```diff
@@ -63,11 +63,11 @@ impl UserNotificationSettingsRepo {
 
     pub async fn find_for_user_ids(
         &self,
-        user_ids: impl IntoIterator<Item = &GaloyUserId>,
+        user_ids: &[GaloyUserId],
```
Comment (bodymindarts) [inline core/notifications/src/user_notification_settings/repo.rs:66]: Why was this changed? The original was broader in what it accepts.
